### PR TITLE
Replace python script with tracegen

### LINKED ISSUE #4503: [spm] Change README to use tracegen
The [README](https://github.com/jaegertracing/jaeger/blob/main/docker-compose/monitor/README.md#setup) currently suggests running `./otlp_exporter_example.py`, why is that?  Can we instead provide an example command of running the `tracegen`?  Instead of saying "[run] a number of times", tracegen can have an exact command line that runs for a specified period of time with given throughput, which seems more useful for SPM. And IIRC we already use it in the docker-compose file, so I would retire the python script altogether.

### PR DESCRIPTION
**Pull request by albertteoh**
## Which problem is this PR solving?
- Resolves #4503

## Description of the changes
- Replaces the python example with tracegen in order to reduce maintenance needed on the script.

## How was this change tested?
- Ran locally E2E to confirm traces and metrics are appearing in Jaeger UI.

## Checklist
- [x] I have read https://github.com/jaegertracing/jaeger/blob/master/CONTRIBUTING_GUIDELINES.md
- [x] I have signed all commits
~- [] I have added unit tests for the new functionality~
- [x] I have run lint and test steps successfully
  - for `jaeger`: `make lint test`
  - for `jaeger-ui`: `yarn lint` and `yarn test`

### CODE REVIEW

File: docker-compose/monitor/README.md [hunk 1]
```diff
@@ -77,38 +77,43 @@ For each "run" make target, you should expect to see the following in the Monito
...
+docker compose up
 ```
 
-Then run this example a number of times to generate some traces: 
+Generate a specific number of traces with:
+```shell
+docker run --env OTEL_EXPORTER_OTLP_TRACES_ENDPOINT="http://otel_collector:4317" \
+  --network monitor_backend \
+  --rm \
+  jaegertracing/jaeger-tracegen:1.49 \
```
Comment (albertteoh) [inline docker-compose/monitor/README.md:94]: Let me know if there's a preference to use the `latest` tag.

**Review by yurishkuro — APPROVED** [review]

**Comment by yurishkuro** [comment]
you can also remove the respective section in .github/dependabot.yml

File: docker-compose/monitor/docker-compose.yml [hunk 2]
```diff
@@ -22,8 +22,6 @@ services:
     volumes:
       - ${OTEL_CONFIG_SRC:-./otel-collector-config-connector.yml}:/etc/otelcol/otel-collector-config.yml
     command: --config /etc/otelcol/otel-collector-config.yml
-    ports:
-      - "4317:4317"
```
Comment (albertteoh) [inline docker-compose/monitor/docker-compose.yml:26]: Note that I've also removed this as we're executing tracegen in the same `monitor_backend` network. Also tested this locally to make sure:

```shell
# Fail to access the port from host machine.
$ telnet localhost 4317
Trying 127.0.0.1...
telnet: Unable to connect to remote host: Connection refused

# Successfully write to port from docker command.
$ docker run --env OTEL_EXPORTER_OTLP_TRACES_ENDPOINT="http://otel_collector:4317" \
  --network monitor_backend \
  --rm \
  jaegertracing/jaeger-tracegen:1.49 \
    -trace-exporter otlp-grpc \
    -traces 1
...
2023-09-12T20:40:01.903Z	LEVEL(-8)	global/internal_logging.go:62	exporting spans	{"count": 2, "total_dropped": 0}
```

**Comment by albertteoh** [comment]
> you can also remove the respective section in .github/dependabot.yml

Nice, thanks. Removed in: [b850f02](https://github.com/jaegertracing/jaeger/pull/4753/commits/b850f023ca75d057f2004f5b2fa237917a707fd5)

**Pull request merged**